pkg/custom-plugins/component.go

```diff
@@ -124,13 +124,35 @@ func (c *component) Check() components.CheckResult {
 	cr.reason = "ok"
 	log.Logger.Debugw("successfully executed plugin", "exitCode", cr.exitCode, "output", string(cr.out))
 
+	// either custom parser (jsonpath) or default parser
+	if len(cr.out) > 0 && c.spec.HealthStatePlugin.OutputParse != nil {
+		var matchResults map[string]extractedExtraInfo
+		matchResults, cr.err = c.spec.HealthStatePlugin.OutputParse.extractExtraInfo(cr.out)
+		if cr.err != nil {
+			log.Logger.Errorw("error extracting extra info", "error", cr.err)
+			cr.reason = fmt.Sprintf("error extracting extra info -- %s (output: %s)", cr.err, string(cr.out))
+		}
+
+		if len(matchResults) > 0 {
+			for k, result := range matchResults {
+				cr.extraInfo[k] = result.fieldValue
+
+				if !result.matched {
+					cr.health = apiv1.HealthStateTypeUnhealthy
+					cr.reason = fmt.Sprintf("extra info %q has unexpected value %q (based on match rule %q)", k, result.fieldValue, result.rule)
```

Comment:
shall we hide the details? Just give the result: we cannot find target value by the rule(name or id, not the rule content)